src/cloudai/_core/test_template_strategy.py

```diff
@@ -29,44 +29,28 @@ class TestTemplateStrategy:
     Attributes
         system (System): The system schema object.
         install_path (str): Path where the benchmarks are to be installed.
-        env_vars (Dict[str, Any]): Default environment variables.
         cmd_args (Dict[str, Any]): Default command-line arguments.
         default_env_vars (Dict[str, str]): Constructed default environment variables.
         default_cmd_args (Dict[str, str]): Constructed default command-line arguments.
     """
 
     __test__ = False
 
-    def __init__(self, system: System, env_vars: Dict[str, Any], cmd_args: Dict[str, Any]) -> None:
+    def __init__(self, system: System, cmd_args: Dict[str, Any]) -> None:
         """
         Initialize a TestTemplateStrategy instance with system configuration, env variables, and command-line arguments.
 
         Args:
             system (System): The system configuration for the test.
-            env_vars (Dict[str, Any]): Default environment variables.
             cmd_args (Dict[str, Any]): Default command-line arguments.
         """
         self.system = system
         self.install_path = ""
-        self.env_vars = env_vars
         self.cmd_args = cmd_args
-        self.default_env_vars = self._construct_default_env_vars()
+        self.default_env_vars = {}
```

Comment:
Quick search shows that we only update `self.default_env_vars`, but never actually use it. Could you please double check and remove if it is not used?

Reply:
We use `self.default_env_vars`. The `default_env_vars` are used to initialize or update `final_env_vars`, which is then utilized for command generation. We cannot remove `default_env_vars`. I will rename and refactor `default_env_vars` and `default_cmd_args` based on our discussion in follow-up PRs.

Reply:
```
src/cloudai/_core/test_template_strategy.py:        default_env_vars (Dict[str, str]): Constructed default environment variables.
src/cloudai/_core/test_template_strategy.py:        self.default_env_vars = {}
src/cloudai/_core/test_template_strategy.py:        self.default_env_vars.update(system.global_env_vars)
src/cloudai/_core/test_template_strategy.py:        default_env_vars: Dict[str, str],
src/cloudai/_core/test_template_strategy.py:            default_env_vars (Dict[str, str]): The default environment variables.
src/cloudai/_core/test_template_strategy.py:        final_env_vars = default_env_vars.copy()
src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/sleep/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/ucc_test/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/nemo_launcher/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/chakra_replay/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/nccl_test/slurm_command_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
src/cloudai/schema/test_template/nccl_test/kubernetes_json_gen_strategy.py:        final_env_vars = self._override_env_vars(self.default_env_vars, extra_env_vars)
```

Reply:
Got it, updated.